lib/elements/element.py

```diff
@@ -186,15 +189,18 @@ def has_style(self, style_name):
     @property
     @cache
     def stroke_scale(self):
-        svg = self.node.getroottree().getroot()
-        doc_width, doc_height = get_doc_size(svg)
-        # this is necessary for clones, since they are disconnected from the DOM
-        # it will result in a slighty wrong result for zig-zag stitches
-        if doc_width == 0:
-            return 1
-        viewbox = svg.get('viewBox', '0 0 %s %s' % (doc_width, doc_height))
-        viewbox = viewbox.strip().replace(',', ' ').split()
-        return doc_width / float(viewbox[2])
+        node_scale = 1
+        node_transform = str(get_node_transform(self.node))
+        if node_transform is not None:
+            if node_transform.startswith('matrix'):
+                decomposed_transform = findall(r"[-+]?\d*\.\d+|\d+", node_transform)
+                sx = sqrt(float(decomposed_transform[0])**2 + float(decomposed_transform[2])**2)
+                sy = sqrt(float(decomposed_transform[1])**2 + float(decomposed_transform[3])**2)
+                node_scale = (sx + sy) / 2
+            elif node_transform.startswith('scale'):
+                decomposed_transform = findall(r"[-+]?\d*\.\d+|\d+", node_transform)
+                node_scale = (float(decomposed_transform[0]) + float(decomposed_transform[1])) / 2
+        return node_scale
```

Comment:
Hmmmmm.  I wonder if we can get creative here.  What if we do `node_transform.apply_to_point((1, 0))`, then take the magnitude of the resulting vector to get `sx`?  Then repeat for `(0, 1)` to get `sy`.  I think we may be able to get inkex to do all the work for us.  What do you think?

Reply:
Oh, my code only works for inkex 1.0+, but there's a similar function to apply a transform to a point in the old inkex

Reply:
I don't know. Maybe this goes a little bit above my head. ... could you take over?
My code here is working, but it's not really very beautiful. I do admit that :D

Reply:
Sure thing. :)  This is something that will probably make more sense in code than English.